src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeParse.cs

```diff
@@ -1901,24 +1901,48 @@ private static bool GetDayOfNNY(ref DateTimeResult result, scoped ref DateTimeRa
 
             if (order == ORDER_MDY || order == ORDER_YMD)
             {
-                if (SetDateYMD(ref result, raw.year, n1, n2))
-                {
-                    result.flags |= ParseFlags.HaveDate;
-                    return true; // MD + Year
-                }
+                if (SetMD(ref result, raw.year, n1, n2))
+                    return true;
+#if TARGET_BROWSER
+                // if we are parsing the datetime string with custom format then the CultureInfo format `order`
+                // does not matter and DM + Year is also possible for NNY
+                if (GlobalizationMode.Hybrid && SetDM(ref result, raw.year, n1, n2))
+                    return true;
+#endif
             }
             else
             {
-                if (SetDateYMD(ref result, raw.year, n2, n1))
-                {
-                    result.flags |= ParseFlags.HaveDate;
-                    return true; // DM + Year
-                }
+                if (SetDM(ref result, raw.year, n1, n2))
+                    return true;
+#if TARGET_BROWSER
+                if (GlobalizationMode.Hybrid && SetMD(ref result, raw.year, n1, n2))
+                    return true;
+#endif
             }
             result.SetBadDateTimeFailure();
             return false;
         }
 
+        private static bool SetMD(ref DateTimeResult result, int year, int month, int day)
```

Comment:
These two method names don't seem correct. Shouldn't they be `SetYMD`, and `SetYDM`?

Reply:
I think I wanted to underline mostly that these two functions have different order of day and month. But your name is less confusing.